src/loaders/compute_tools/tool_common.py

```diff
@@ -246,46 +226,67 @@ def _get_data_ids(self):
             data_ids = all_data_ids
         return list(set(data_ids))
 
+    def _prepare_execution(
+            self,
+            unzip: bool
+    ) -> Tuple[Path, Dict[str, Dict[str, Union[str, Path]]], List[str]]:
+
+        batch_dir, genomes_meta = _prepare_tool(
+            self._work_dir,
+            loader_common_names.COMPUTE_OUTPUT_NO_BATCH,
+            self._node_id,
+            self._data_ids,
+            self._source_data_dir,
+            self._source_file_ext,
+            self._allow_missing_files,
+            self._tool_data_id_from_filename,
+            self._suffix_ids,
+        )
+
+        unzipped_files_to_delete = []
+        if unzip:
+            unzipped_files_to_delete = _unzip_files(genomes_meta)
+
+        return batch_dir, genomes_meta, unzipped_files_to_delete
+
+    def _finalize_execution(
+            self,
+            unzipped_files_to_delete: List[str],
+            genomes_meta: Dict[str, Dict[str, Union[str, Path]]],
+            batch_dir: Path):
+
+        if unzipped_files_to_delete:
+            print(f"Deleting {len(unzipped_files_to_delete)} unzipped files: {unzipped_files_to_delete[:5]}...")
+            for file in unzipped_files_to_delete:
+                os.remove(file)
+
+        create_metadata_file(genomes_meta, batch_dir)
+
     def parallel_single_execution(self, tool_callable: Callable[[str, str, Path, Path, int, bool], None], unzip=False):
         """
         Run a tool by a single data file, storing the results in a single batch directory with
         the individual runs stored in directories by the data ID.
 
-        One tool execution per data ID. Tool execution is parallelized by the number of threads
-        specified in the constructor.
+        One tool execution per data ID. Tool execution is in a serial manner (utilizing multiprocessing with a constant process count of 1).
         Results from execution need to be processed/parsed individually.
 
         Use case: microtrait - execute microtrait logic on each individual genome file. The result file is stored in
                   each individual genome directory. Parser program will parse the result file in each individual genome
                   directory.
 
-        tool_callable - the callable for the tool that takes 5 arguments:
+        tool_callable - the callable for the tool that takes 6 arguments:
             * The tool safe data ID
             * The data ID
             * The input file
             * The output directory
+            * The number of threads to use for the tool run
```

Comment:
The docs were missing last time when adding threads to single tool runs?

Reply:
yea. I probably missed that.

Reply:
:+1: